.travis.yml

```diff
@@ -60,51 +61,57 @@ matrix:
             - libstdc++6
             - fonts-droid
       before_script:
-        - wget https://dl.google.com/android/repository/sdk-tools-linux-3859397.zip
-        - mkdir android-sdk
-        - unzip -qq sdk-tools-linux-3859397.zip -d android-sdk
+        - ./script/before_build_apks.sh
         - export ANDROID_HOME=`pwd`/android-sdk
-        - export PATH=`pwd`/android-sdk/tools/bin:$PATH
-        - mkdir -p /home/travis/.android # silence sdkmanager warning
-        - echo 'count=0' > /home/travis/.android/repositories.cfg # silence sdkmanager warning
-        # suppressing output of sdkmanager to keep log under 4MB (travis limit)
-        - echo y | sdkmanager "tools" >/dev/null
-        - echo y | sdkmanager "platform-tools" >/dev/null
-        - echo y | sdkmanager "build-tools;26.0.3" >/dev/null
-        - echo y | sdkmanager "platforms;android-26" >/dev/null
-        - echo y | sdkmanager "extras;android;m2repository" >/dev/null
-        - echo y | sdkmanager "extras;google;m2repository" >/dev/null
-        - echo y | sdkmanager "patcher;v4" >/dev/null
-        - sdkmanager --list
-        - wget http://services.gradle.org/distributions/gradle-4.1-bin.zip
-        - unzip -qq gradle-4.1-bin.zip
-        - export GRADLE_HOME=$PWD/gradle-4.1
-        - export PATH=$GRADLE_HOME/bin:$PATH
-        - gradle -v
-        - git clone https://github.com/flutter/flutter.git
         - export PATH=`pwd`/flutter/bin:`pwd`/flutter/bin/cache/dart-sdk/bin:$PATH
-        - flutter doctor
-        - pub global activate flutter_plugin_tools
       script:
         - ./script/plugin_tools.sh build-examples --apk
         - ./script/plugin_tools.sh java-test  # must come after apk build
-    # Job 4) Build example IPAs
+    # Job 3.2) Build example APKs and run Java tests, shard 2/2
```

Comment:
I would have liked to avoid duplication here, but not sure what is the best approach for that.